pkg/files/file_helpers.go

```diff
@@ -7,35 +7,37 @@
 package files
 
 import (
+	"bytes"
 	"cmp"
-	"crypto/sha256"
-	"encoding/base64"
 	"fmt"
 	"io"
 	"os"
 	"slices"
+	"strconv"
 
-	"github.com/nginx/agent/v3/api/grpc/mpi/v1"
-	"github.com/nginx/agent/v3/pkg/uuid"
+	"github.com/google/uuid"
+	mpi "github.com/nginx/agent/v3/api/grpc/mpi/v1"
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-func GetFileMeta(filePath string) (*v1.FileMeta, error) {
+const permissions = 0o644
+
+func GetFileMeta(filePath string) (*mpi.FileMeta, error) {
 	fileInfo, err := os.Stat(filePath)
 	if err != nil {
 		return nil, err
 	}
 
-	hash, err := GenerateFileHash(filePath)
+	_, hash, err := GenerateHashWithReadFile(filePath)
 	if err != nil {
 		return nil, err
 	}
 
-	return &v1.FileMeta{
+	return &mpi.FileMeta{
 		Name:         filePath,
 		Hash:         hash,
 		ModifiedTime: timestamppb.New(fileInfo.ModTime()),
-		Permissions:  fileInfo.Mode().Perm().String(),
+		Permissions:  "0777",
```

Comment:
Is this meant to be hardcoded to `0777`?

Reply:
No will change it back